esp/services/ws_workunits/ws_workunitsHelpers.cpp

```diff
@@ -1859,19 +1859,9 @@ void appendIOStreamContent(MemoryBuffer &mb, IFileIOStream *ios, bool forDownloa
     while (!eof)
     {
         line.clear();
-        for (;;)
-        {
-            char c;
-            size32_t numRead = ios->read(1, &c);
-            if (!numRead)
-            {
-                eof = true;
-                break;
-            }
-            line.append(c);
-            if (c=='\n')
-                break;
-        }
+        eof = ios->readLine(true, line);
+        if (line.isEmpty())
```

Comment:
I'm not sure this should be possible.
If empty, eof should be set I think.

Reply:
Yes with the IStreamLineReader.